src/nat/middleware/cache_middleware.py

```
@@ -0,0 +1,254 @@
+        self._cache: dict[str, Any] = {}
+
+    def _should_cache(self) -> bool:
+        """Check if caching should be enabled based on the current context."""
+        if self._enabled_mode == "always":
+            return True
+
+        # Get the current context and check if we're in evaluation mode
+        try:
+            context_state = ContextState.get()
+            context = Context(context_state)
+            return context.is_evaluating
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)